sensei/Catalyst2AnalysisAdaptor.cxx

```diff
@@ -0,0 +1,180 @@
+#include "MeshMetadata.h"
+#include "Profiler.h"
+#include "SVTKUtils.h"
+
+#include <catalyst.h>
+#include <catalyst_conduit.hpp>
+
+#include <svtkDataObject.h>
+#include <svtkCompositeDataSetRange.h>
+#include <svtkDataObjectToConduit.h>
+#include <svtkDataSet.h>
+#include <svtkMultiBlockDataSet.h>
+#include <svtkPointData.h>
+#include <svtkRange.h>
+
+namespace sensei
+{
+
+// #define Catalyst2DebugMacro(x) std::cerr x << std::endl;
+#define Catalyst2DebugMacro(x)
+
+//-----------------------------------------------------------------------------
+senseiNewMacro(Catalyst2AnalysisAdaptor);
+
+//-----------------------------------------------------------------------------
+Catalyst2AnalysisAdaptor::Catalyst2AnalysisAdaptor()
+{
+  this->Initialize();
+}
+
+//-----------------------------------------------------------------------------
+Catalyst2AnalysisAdaptor::~Catalyst2AnalysisAdaptor()
+{
+  this->Finalize();
+}
+
+//-----------------------------------------------------------------------------
+void Catalyst2AnalysisAdaptor::Initialize() {}
+
+//-----------------------------------------------------------------------------
+void Catalyst2AnalysisAdaptor::AddPythonScriptPipeline(const std::string& fileName)
+{
+  conduit_cpp::Node node;
+  node["catalyst/scripts/script"].set_string(fileName);
+  auto error_code = catalyst_initialize(conduit_cpp::c_node(&node));
+  if (error_code != catalyst_status_ok)
+  {
+    // you are in trouble young man
+    std::cerr << "catalyst initialize failed with code: " << error_code << std::endl;
+  }
+}
+
+//-----------------------------------------------------------------------------
+void Catalyst2AnalysisAdaptor::SetMeshes(const std::vector<std::string>& a)
+{
+  std::copy(a.begin(),
+            a.end(),
+            std::inserter(this->meshes, this->meshes.end()));
+}
+
+//----------------------------------------------------------------------------
+bool Catalyst2AnalysisAdaptor::Execute(DataAdaptor* dataAdaptor, DataAdaptor**)
+{
+  TimeEvent<128> mark("Catalyst2AnalysisAdaptor::Execute");
+
+  // Get a description of the simulation metadata
+  unsigned int nMeshes = 0;
+  if (dataAdaptor->GetNumberOfMeshes(nMeshes))
+  {
+    SENSEI_ERROR("Failed to get the number of meshes")
+      return false;
+  }
+
+  std::vector<MeshMetadataPtr> metadata(nMeshes);
+  for (unsigned int i = 0; i < nMeshes; ++i)
+  {
+    MeshMetadataPtr mmd = MeshMetadata::New();
+
+    if (dataAdaptor->GetMeshMetadata(i, mmd))
+    {
+      SENSEI_ERROR("Failed to get metadata for mesh " << i << " of " << nMeshes)
+        return false;
+    }
+    metadata[i] = mmd;
+  }
+
+  // Conduit node to fill
+  conduit_cpp::Node exec_params;
+  int timeStep = dataAdaptor->GetDataTimeStep();
+
+  for (auto meta : metadata)
+  {
+    const char* meshName = meta->MeshName.c_str();
+    // get the mesh only if it specified or no meshes are specified in the XML
+    if (this->meshes.empty() || this->meshes.find(meshName) != this->meshes.end())
```

Comment:
This solution only handle mesh but still pulls all the arrays which is bad for in transit. You should use [sensei::DataRequirements](https://github.com/SENSEI-insitu/SENSEI/blob/master/sensei/DataRequirements.h) which has all the XML parsing code for meshes, and point and cell centered arrays. All analysis adaptors that need to have manual specification use this approach, which makes the XML consistent between the different implementations. For instance see how [sensei::ConfigurableAnalysis initializes the sensei::VTKPosthocIO analysis](https://github.com/SENSEI-insitu/SENSEI/blob/294f5c89618147a065efc36a9fd7c05ee823d0a6/sensei/ConfigurableAnalysis.cxx#L869) and how the [sensei::VTKPosthocIO iterates over the specified meshes and arrays](https://github.com/SENSEI-insitu/SENSEI/blob/294f5c89618147a065efc36a9fd7c05ee823d0a6/sensei/VTKPosthocIO.cxx#L285).

Reply:
Great. Sounds good.